File: example-android-project/app/src/main/java/io/github/gofaith/faithdroid/FViews/FSnackbar.java
package io.github.gofaith.faithdroid.FViews;

import android.support.design.widget.Snackbar;
import android.view.View;

import org.json.JSONArray;
import org.json.JSONTokener;

import faithdroid.Faithdroid;
import io.github.gofaith.faithdroid.UI.AttrGettable;
import io.github.gofaith.faithdroid.UI.AttrSettable;
import io.github.gofaith.faithdroid.UI.UIController;

public class FSnackbar extends FView implements AttrSettable, AttrGettable {
    public Snackbar v;

    public FSnackbar(UIController controller, View anchorView) {
        parentController=controller;
        v = Snackbar.make(anchorView, "", Snackbar.LENGTH_SHORT);
    }
    @Override
    public String getAttr(String attr) {
        return null;
    }

    @Override
    public void setAttr(String attr, String value) {
        if (value == null) {
            return;
        }
        switch (attr) {
            case "Text":
                v.setText(value);
                break;
            case "Action":
                try {
                    final JSONArray array = (JSONArray) (new JSONTokener(value).nextValue());
                    final String fnId=array.getString(1);
                    v.setAction(array.getString(0),
                            new View.OnClickListener() {
                                @Override
                                public void onClick(View v) {
                                    Faithdroid.triggerEventHandler(fnId, "");
                                }
                            });
                } catch (Exception e) {
                    e.printStackTrace();
                }
                break;
            case "Show":
                v.show();
                break;
        }
    }
}

File: android/app/src/main/java/io/github/stevenzack/faithdroidapp/FViews/FSnackbar.kt
package io.github.stevenzack.faithdroidapp.FViews

import android.view.View
import com.google.android.material.snackbar.Snackbar
import faithdroid.Faithdroid
import io.github.stevenzack.faithdroidapp.UI.AttrGettable
import io.github.stevenzack.faithdroidapp.UI.AttrSettable
import io.github.stevenzack.faithdroidapp.UI.FView
import io.github.stevenzack.faithdroidapp.UI.UIController
import org.json.JSONArray
import org.json.JSONTokener


class FSnackbar(controller: UIController, anchorView: View) : FView(), AttrSettable, AttrGettable {
    var v: Snackbar

    init {
        parentController = controller
        v = Snackbar.make(anchorView, "", Snackbar.LENGTH_SHORT)
    }

    override fun getAttr(attr: String): String {
        return ""
    }

    override fun setAttr(attr: String, value: String) {
        if (value == null) {
            return
        }
        when (attr) {
            "Text" -> v.setText(value)
            "Action" -> try {
                val array = JSONTokener(value).nextValue() as JSONArray
                val fnId = array.getString(1)
                v.setAction(array.getString(0),
                    View.OnClickListener { Faithdroid.triggerEventHandler(fnId, "") })
            } catch (e: Exception) {
                e.printStackTrace()
            }

            "Show" -> v.show()
        }
    }
}
